### PR TITLE
Implemented student view component

### LINKED ISSUE #94: Create Student View API According to API
student-view 1

### PR DESCRIPTION
**Pull request by richardzeng1**
fixes #94

### CODE REVIEW

File: student_survey.php [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+                exit();
+            case "POST":
+                print json_encode(handle_post($_POST));
+                exit();
+            default:
+                throw new Exception("Invalid Request");
+        }
+    } else {
+        $error = 'No Request Method Found';
+        throw new Exception($error);
```
Comment (siefkenj) [inline student_survey.php]: Why is a method needed?

File: student_survey.php [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+            $fetched = $stmt->fetchAll(PDO::FETCH_ASSOC);
+            return $fetched;
+        } else {
+            return array("TYPE" => "Success");
+        }
+    } catch (PDOException $e) {
+        $result = set_http_response(500);
+        date_default_timezone_set('America/Toronto');
+        error_log(
+            date("Y-m-d h:i:sa") . " : " . $e->getMessage() . "\n",
```
Comment (siefkenj) [inline student_survey.php]: You need `\n` in the error log?

File: student_survey.php [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+    if (isset($body['question_responses']) && $body['question_responses']) {
+        $question_responses = $body['question_responses'];
+    }
+
+    // Removing start and end square brackets
+    $question_responses = rtrim($question_responses, "]");
+    $question_responses = ltrim($question_responses, "[");
+
+    // Putting each JSON as an element in $matches
+    preg_match_all('/{(.*?)}/', $question_responses, $matches);
```
Comment (siefkenj) [inline student_survey.php]: This is errorprone. Parse the JSON and then get what you need from that.

File: student_survey.php [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+    foreach ($responses as $response) {
+        $bind_variables[":question_id"] = $response->question_id;
+        $bind_variables[":answer"] = $response->response;
+
+        array_push(
+            $return_val_data,
+            get_query_result(gen_query_submit_responses(), $bind_variables)
+        );
+    }
+    return array("TYPE" => "Success", "DATA" => $return_val_data);
```
Comment (siefkenj) [inline student_survey.php]: `"success"`

File: student_survey.php (hunk 1 above)
Comment (richardzeng1) [reply student_survey.php]: It's used when `$_SERVER['REQUEST_METHOD']` is not found. That means that a GET or POST request is not specified. It's the boilerplate code used from get_info.php.

File: student_survey.php (hunk 2 above)
Comment (richardzeng1) [reply student_survey.php]: Yes or else every error message would display on the same line in errors.log. It's a carry over from get_info.php.

**Pull request merged**